src/BuiltInTools/dotnet-watch/HotReload/NamedPipeContract.cs

```diff
@@ -19,58 +29,32 @@ internal readonly struct UpdatePayload(IReadOnlyList<UpdateDelta> deltas, Respon
         /// </summary>
         public async ValueTask WriteAsync(Stream stream, CancellationToken cancellationToken)
         {
-            await using var binaryWriter = new BinaryWriter(stream, Encoding.UTF8, leaveOpen: true);
-            binaryWriter.Write(Version);
-            binaryWriter.Write(Deltas.Count);
+            await stream.WriteAsync(Version, cancellationToken);
+            await stream.WriteAsync(Deltas.Count, cancellationToken);
 
-            for (var i = 0; i < Deltas.Count; i++)
+            foreach (var delta in Deltas)
             {
-                var delta = Deltas[i];
-                binaryWriter.Write(delta.ModuleId.ToString());
-                await WriteBytesAsync(binaryWriter, delta.MetadataDelta, cancellationToken);
-                await WriteBytesAsync(binaryWriter, delta.ILDelta, cancellationToken);
-                await WriteBytesAsync(binaryWriter, delta.PdbDelta, cancellationToken);
-                WriteIntArray(binaryWriter, delta.UpdatedTypes);
+                await stream.WriteAsync(delta.ModuleId, cancellationToken);
```

Comment:
Are there places in here where the state of the CancallationToken should be checked?

Reply:
You mean explicitly calling cancellationToken.ThrowIfCancellationRequested? There is no expensive work done in these helpers they are just calling stream.WriteAsync